WordPress/gutenberg
# [RNMobile] Navigation in bottom-sheet

## Description
Main issue: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1884
Wordpress-iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/14474
Wordpress-Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/12455
Gutenberg-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2549
In this PR i added `react-navigation` and used it inside our Bottom-Sheet. This Pr has a long story behind. We wanted to replace the entire Bottom-Sheet to `reanimated-bottom-sheet` but unfortunately, there are too many limitations so we decided to keep our current BottomSheet implementation and just add navigation inside.

This PR affects all Block Settings. 

I implemented some helper mechanisms that will be useful in the future while adding a new screen inside the bottom-sheet. To animate the bottom-sheet height when the navigation is inside we need to set the height of current Screen ( The visible one If we have nested navigators) so I added BottomSheet.NavigationScreen that handles the Android hardware back button and BottomSheet.NavigationContainer to animate height of it.

## How has this been tested?
- Check settings of each block
- Open Button Block settings
- Click on Text Color / Backgrounds color
- The navigation inside bottom sheet should be possible
- Play a bit with colors and check if the button reflecting these changes
- Try the hardware back button on Android - the back button should back to the previous screen or hide the bottom-sheet if it's the main settings screen

## Screenshots 
iOS | Android
---|---
![bottom-sheet-navigation-finish](https://user-images.githubusercontent.com/16336501/87296613-50d80a00-c507-11ea-806d-6698193e50a3.gif) | ![andoridbottomsheetbuttonfinish](https://user-images.githubusercontent.com/16336501/87306455-5558ee80-c518-11ea-9970-11fd77c71477.gif)


## Types of changes
Add 'react-navigation' to the project and add possibilities to use it inside our bottom-sheet.

## Checklist:
